torch/ao/nn/sparse/linear.py

```diff
@@ -0,0 +1,32 @@
+import torch
+from torch.nn import functional as F
+
+class Linear(torch.nn.Linear):
+    mask: torch.Tensor
+    inplace_masking: bool
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        mask = torch.ones(self.weight.shape)
+        self.register_buffer('mask', mask)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        return F.linear(x, self.weight_pre_hook(), self.bias)
+
+    def weight_pre_hook(self):
+        return self.weight * self.mask
+
+    def _get_name(self):
+        return 'SparseLinear'
+
+    def load_state_dict(self, state_dict, strict=True):
+        r"""We allow loading from both sparse and dense layers."""
+        if 'mask' not in state_dict:
+            state_dict['mask'] = torch.ones(state_dict['weight'].shape)
+        return super().load_state_dict(state_dict, strict=strict)
+
+    @classmethod
+    def from_dense(cls, dense):
+        sparse = cls(dense.in_features, dense.out_features, (dense.bias is not None))
```

Comment:
Can we handle this instead by doing the default init of the mask attr to be all ones? This way from_dense is not needed.